src/libraries/System.Text.Json/tests/JsonDocumentTests.cs

```diff
@@ -3650,7 +3647,7 @@ private static string GetCompactJson(TestCaseType testCaseType, string jsonStrin
             return s_compactJson[testCaseType] = existing;
         }
 
-        [Fact]
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsThreadingSupported))]
```

Comment:
It wouldn't be testing the exact same thing, but it seems like this test would "just work" if you changed the `Task.WaitAll(tasks)` to instead be `await Task.WhenAll(tasks)` and made the method async, no?

Reply:
Yeah that does indeed work. I can send a followup PR but I'd like to first get this one in while CI is green :)